### PR TITLE
ChakraCore 2018-05 security updates

### PR DESCRIPTION
**Pull request by MSLaguana**

### CODE REVIEW

**Comment by MSLaguana** [comment]
cc @meg-gupta @sigatrev @pleath @sethbrenith @Penguinwizzard @rajatd @akroshg @agarwal-sandeep

**Comment by akroshg** [comment]
LGTM

**Comment by sigatrev** [comment]
:shipit:

**Comment by sethbrenith** [comment]
LGTM

File: lib/Runtime/Library/ScriptFunction.cpp [hunk 1]
```diff
@@ -83,50 +83,47 @@ namespace Js
...
-            return pfuncScriptWithInlineCache;
         }
-        else
+
+        if (pfuncScript == nullptr)
         {
-            ScriptFunction* pfuncScript = scriptContext->GetLibrary()->CreateScriptFunction(functionProxy);
-            pfuncScript->SetEnvironment(environment);
+            pfuncScript = scriptContext->GetLibrary()->CreateScriptFunction(functionProxy);
+        }
```
Comment (pleath) [inline lib/Runtime/Library/ScriptFunction.cpp]: This is the part of the change that requires additional logic for release/1.9+. We can synch up to manage that.

**Pull request merged**